### DISCUSSION TITLE
Posting notifications on ft.IconButton

### DISCUSSION THREAD

**Discussion by Aidar1983** [flet-dev/flet#2750] [Q&A] [answered]
### Question

Good afternoon! how to place ft.Badge inside the ft.Icon Button, this is necessary to show the presence of notifications about the presence of unread messages. I'm trying to do this, but it doesn't work. 

page.appbar = ft.CupertinoAppBar(
        leading=ft.IconButton(ft.icons.ASSIGNMENT_IND, icon_color="white", on_click=show_drawer),
        bgcolor=ft.colors.DEEP_ORANGE_ACCENT_200,
        trailing=ft.IconButton(ft.icons.DOORBELL, icon_color="white",content=ft.Badge(text="1")))

### Code sample

```python
page.appbar = ft.CupertinoAppBar(
        leading=ft.IconButton(ft.icons.ASSIGNMENT_IND, icon_color="white", on_click=show_drawer),
        bgcolor=ft.colors.DEEP_ORANGE_ACCENT_200,
        trailing=ft.IconButton(ft.icons.DOORBELL, icon_color="white",content=ft.Badge(text="1")))
```

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by InesaFitsner** [comment, marked as answer]
To display Badge on the IconButton you need to do the opposite: use IconButton as the content for the Badge: 
```
import flet as ft


def main(page: ft.Page):
    page.appbar = ft.CupertinoAppBar(
        leading=ft.IconButton(ft.icons.ASSIGNMENT_IND, icon_color="white"),
        bgcolor=ft.colors.DEEP_ORANGE_ACCENT_200,
        trailing=ft.Badge(
            content=ft.IconButton(
                ft.icons.DOORBELL,
                icon_color="white",
            ),
            text="1",
        ),
    )
    page.update()


ft.app(target=main)

```

**Comment by Aidar1983** [comment]
It's working! Thank you very much
